packages/db/src/translate/rules.ts

```diff
@@ -6,27 +6,47 @@ import { CollectionPgModel } from '../models/collection';
 import { ProviderPgModel } from '../models/provider';
 import { PostgresRule, PostgresRuleRecord } from '../types/rule';
 
+// Rule queries can include fields from external tables like provider and collection
+interface RuleRecordWithExternals extends PostgresRuleRecord {
+  provider?: string,
+  collectionName?: string,
+  collectionVersion?: string,
+}
+
 export const translatePostgresRuleToApiRule = async (
-  pgRule: PostgresRuleRecord,
+  pgRule: RuleRecordWithExternals,
```

Comment:
I wanted to use the existing translate method, not a new one that's mostly a copy but without DB queries. To do that though the types get a little weird. Open to feedback.

Reply:
I like the approach and agree that we shouldn't make near-dupe functions if we can get away with it.
*but* I don't like the idea that this function can and is intended to be used without ever actually calling 3/4 of its parameters.
I foresee outside functions having to pass around knex and PGmodels for no reason to satisfy this

Reply:
is there a sensible way to write this so that knex is optional?
catch if we need to call it and didn't instantiate it and throw an error so a dev knows they misunderstood?

Reply:
Yeah, that's fair. Maybe we do a separate method for translating without the db calls but we call that method from the existing `translatePostgresRuleToApiRule`. So You have `translatePostgresRuleToApiRule` that requires a knex object and calls something like `translatePostgresRuleToApiRuleWithoutExternals`. Then we could use `translatePostgresRuleToApiRuleWithoutExternals` separately as necessary.

I'll refactor a bit and let ya know. Feel free to disagree with the approach. Happy to chat.

Reply:
Oh ha you're right. `translatePostgresGranuleToApiGranule` calls `translatePostgresGranuleToApiGranuleWithoutDbQuery`. Didn't think about that when I replied to Ethan's original comment.

So yes, the original idea was to avoid that and just have one method. The downside is that you're passing a knex object and other params that you don't need in this case. And it's a bit more complex. The upside is that it's only one method.

Maybe I should just do what you all did for granules and executions because I accidentally arrived at the same conclusion and I don't think we want rules to be a unicorn.

Reply:
I've finished reinventing the wheel you invented earlier: https://github.com/nasa/cumulus/pull/3760/commits/fd8f5bfdefec61549adc5ded6c396fcf5e3dbc8c